SpatialGDK/Documentation/content/non-unreal-server-worker-types.md

```diff
@@ -13,33 +13,52 @@ In order to interact with each other, Unreal and non-Unreal server-worker instan
+
 ### How to set up interaction
-To set up your game to interact with SpatialOS components defined outside the GDK (external SpatialOS components), you use the following:
+To set up your game to interact with SpatialOS components defined outside the GDK (external SpatialOS components), you have two options:
 
-* To send SpatialOS component updates and commands, use methods defined in the `SpatialWorkerConnection.h` file (described in the examples section below).
-* To receive [network operations](https://docs.improbable.io/reference/latest/shared/design/operations) for external SpatialOS components, you must provide custom callbacks for specific component IDs and operation types. The GDK then forwards the operations to your callbacks. (This is described in the examples section below.)
+* _(recommended)_ Using the GDK `ExternalSchemaCodegen.bat` script to generate classes for external schema types and an interface for sending updates and registering callbacks for receiving updates.
 
+* Sending updates and registering callback for receiving updates directly through the GDK, requiring you to [serialize and deserialize the low-level C API types](https://docs.improbable.io/reference/latest/capi/serialization).
 
->**TIP:** If you’re using schema from outside the GDK, you can customise [snapshot generation](https://docs.improbable.io/unreal/alpha/content/generating-a-snapshot#generating-a-snapshot) from the GDK toolbar’s  **Snapshot** button. This is to serialize additional entities with these external components which the default Unreal snapshot generation cannot currently do. See [Add to the snapshot](#add-to-the-snapshot) below for how to do this.
+>**Note:** Your external SpatialOS components must have an ID between 1000 and 2000 to be registered by the pipeline.
+
+#### Using the code generation
+
+The external schema code generation tool generates C++ code in your game module from a specific directory containing schema files. The generated code includes classes representing each external schema enumeration and type, and an `ExternalSchemaInterface` class for sending updates and registering callbacks for receiving network operations. The `ExternalSchemaInterface` generates methods for each component and command within a component in your external schema.
+
+* For each component, the `ExternalSchemaInterface` will contain methods for `SendComponentUpdate`, `OnAddComponent`, `OnComponentUpdate`, `OnAuthorityChange`, and `OnRemoveComponent`.
+* For each command, the `ExternalSchemaInterface` will contain methods for `SendCommandRequest`, `SendCommandResponse`, `OnCommandRequest` and `OnCommandResponse`.
+
+##### Setting up
 
-#### Send data
-You set up your game to send SpatialOS component updates, command requests, and command responses directly using the `SpatialWorkerConnection.h` public methods:
+To interact with the external schema types by using the code generation tool, you'll need to do the following:
```

Comment:
I saw there's a detailed explanation of these steps with examples below. Do you think it's worth mentioning this here or linking for each bullet point `(see example)`?